report/report.go

```diff
@@ -24,13 +24,13 @@ const (
 // Reporter is responsible for reporting the profiling report to the destination.
 type Reporter interface {
 	// ReportCPUProfile sends the CPU profiling data to the specific destination.
-	ReportCPUProfile(ctx context.Context, r io.Reader, ci CPUInfo) error
+	ReportCPUProfile(ctx context.Context, r io.Reader, size int, ci CPUInfo) error
```

Comment:
```suggestion
	ReportCPUProfile(ctx context.Context, r io.Reader, fileSize int, ci CPUInfo) error
```

A more explicit name is better, I think. `size` leads the reader to be confused about what size is for.

Reply:
Is it necessary to expose the size argument to the interface? How about just accepting an io.Reader as before, and using io.TeeReader or io.Copy to measure the size internally?

Reply:
> How about just accepting an io.Reader as before, and using io.TeeReader or io.Copy to measure the size internally?

Sounds good.

Reply:
I originally made the file size a parameter because I was concerned about excessive memory usage with large amounts of data being read using `io.Reader`. However, if we can guarantee that the data passed via io.Reader is small enough to be safely loaded into memory, it would actually be better to use the []byte type instead.

Meanwhile, winterjung mentioned in the office that making breaking changes to the interface could cause issues for places where Reporter is used directly. I thought providing a migration guide for those cases would be sufficient, but I’d like to know if there are any additional concerns

Reply:
@mingrammer Gentle reminder.
IMHO it could be better to avoid breaking changes.

Reply:
Profile file size is up to a couple of megabytes, so we don't have to worry about loading into memory. Keeping the interface and calculating the size in Reporter implementation would be a better choice.